core/src/main/scala/cats/data/NonEmptyList.scala

```diff
@@ -17,15 +17,54 @@ final case class NonEmptyList[+A](head: A, tail: List[A]) {
 
   /**
    * Return the head and tail into a single list
+   * {{{
+   * scala> import cats.data.NonEmptyList
+   * scala> val nel = NonEmptyList.of(1, 2, 3, 4, 5)
+   * scala> nel.toList
+   * res0: scala.collection.immutable.List[Int] = List(1, 2, 3, 4, 5)
+   * }}}
    */
   def toList: List[A] = head :: tail
 
-  def last: A = tail.lastOption match {
-    case None    => head
-    case Some(a) => a
+  /**
+   * Selects the last element
+   * {{{
+   * scala> import cats.data.NonEmptyList
+   * scala> val nel = NonEmptyList.of(1, 2, 3, 4, 5)
+   * scala> nel.last
+   * res0: Int = 5
+   * }}}
+   */
+  def last: A = tail.lastOption.getOrElse(head)
+
```

Comment:
Please don't change this implementation. The one above avoids the allocation of the closure and was the motivation for writing it that way.